### ISSUE TITLE
CommandError: Database couldn't be flushed when application model is overridden

### ISSUE THREAD

**Issue by poswald** [evonove/django-oauth-toolkit#143]
In the case where an application model is swapped out, I believe the application table should not be created in the database. The initial migration currently is unaware of this and creates it in both cases.

The reason this causes a problem is if you use a `TransactionTestCase` then django-admin `sqlflush` command is used to truncate the tables. If you add oauth2_provider application to a project which has `TransactionTestCase` unit tests and also overrides the `OAUTH2_PROVIDER_APPLICATION_MODEL` setting with a new application model, then all of those unit tests will fail because the migration is creating a table that django thinks is swapped out.

Here we can see that the table is swapped and managed:
```
In [9]: from django.db.models import get_model
In [10]: get_model("oauth2_provider", "application")._meta.swapped
Out[10]: 'integrations.MyApplication'
In [11]: get_model("oauth2_provider", "application")._meta.managed
Out[11]: True
```

And here's the error from a unit test that fails:

```
CommandError: Database test_myproj couldn't be flushed. Possible reasons:
  * The database isn't running or isn't configured correctly.
  * At least one of the expected database tables doesn't exist.
  * The SQL was invalid.
Hint: Look at the output of 'django-admin.py sqlflush'. That's the SQL this command wasn't able to run.
The full error: cannot truncate a table referenced in a foreign key constraint
DETAIL:  Table "oauth2_provider_application" references "auth_user".
HINT:  Truncate table "oauth2_provider_application" at the same time, or use TRUNCATE ... CASCADE.
```

I _believe_ the fix should be to only issue the create in the migration if the table isn't overridden.

**Comment by JamesPiechota** [comment]
I hit this as well and deleting oauth2_provider_application in a new South schemamigration seems to be working (only done initial tests so far, so no telling if something will bite me later).

For example, in my app (the one that extends the Application model), I did:
`python manage.py schemamigration myapp drop_oauth2_provider_application --empty --freeze oauth2_provider`

And then defined the migration like:
```python
[...]
    depends_on = (
        ("oauth2_provider", "0002_adding_indexes"),
    )

    def forwards(self, orm):
        db.delete_table('oauth2_provider_application', cascade=True)
        
    def backwards(self, orm):
        pass
[...]
```

**Comment by poswald** [comment]
I believe this is now fixed in 0.10.0 due to this [change in the initial migration](https://github.com/evonove/django-oauth-toolkit/commit/9895b76fa2d739302a5e904997653a31d03eefb1)
